[LENS-979] Updates the README

npushkarskii opened the pull request:
This PR suggests updates for the README file.

The updates aim for improving formatting and readability.

btaillon commented on README.md:
```diff
@@ -1,58 +1,66 @@
 # Push API Client
 
-A Coveo Push API Client in Java
+A [Coveo Push API ](https://docs.coveo.com/en/12/api-reference/push-api) client library for Java.
```
```suggestion
A [Coveo Push API](https://docs.coveo.com/en/12/api-reference/push-api) client library for Java.
```

may be a good idea to run a formatter on this. E.g.,
```bash
prettier --write README.md
```

btaillon commented on README.md:
```diff
@@ -1,58 +1,66 @@
-### Step 1: Prerequisites
-The Coveo `push-api-client.java` package is stored on Github packages. You will need a personal access token (classic) with at least `read:packages` scope to install this dependency.
+### Step 1: Update `settings.xml`
+You can install this GitHub Package with [Apache Maven by](https://maven.apache.org/) editing the `~/.m2/settings.xml` file:
```
```suggestion
You can install this GitHub Package with [Apache Maven](https://maven.apache.org/) by editing the `~/.m2/settings.xml` file:
```

btaillon commented on README.md:
````diff
@@ -1,58 +1,66 @@
-### Step 2: Update `settings.xml`
-You can install this GitHub Package with Apache Maven by editing your `~/.m2/settings.xml`:
+    ```xml
+    <repository>
+    <id>github</id>
+    <url>https://maven.pkg.github.com/coveo/push-api-client.java</url>
+    <snapshots>
+        <enabled>true</enabled>
+    </snapshots>
+    </repository>
````
```suggestion
    <repository>
        <id>github</id>
        <url>https://maven.pkg.github.com/coveo/push-api-client.java</url>
        <snapshots>
            <enabled>true</enabled>
        </snapshots>
    </repository>
```

npushkarskii commented:
@btaillon Thx a lot for review, Benjamin! I've addressed your comments and run the prettier.

Feel free to revisit this PR to approve/comment/disapprove 😇

y-lakhdar commented on README.md:
````diff
@@ -82,45 +91,56 @@ public class PushOneDocument {
-    </Loggers>
-</Configuration>
-```
+If you want to push multiple documents to your Coveo organization and use a service for that (e.g. `PushService`, `StreamService`), you need to configure a **logger** to be able to see what happens.
````
It is not required to configure a logger. The logger is useful to catch error messages/warnings from the SDK

npushkarskii replied:
@y-lakhdar thx a lot! I've updated the wording here:

> If you want to push multiple documents to your Coveo organization and use a service for that (e.g. `PushService`, `StreamService`), you may find it useful to configure a **logger** to catch error and warning messages.

sebalvarez07 reviewed (approved):
Looks good on my side!
